odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1160 @@
+        elif isinstance(other, Functional):
+            return FunctionalSum(self, other)
+        else:
+            return NotImplemented
+
+    # Since addition is commutative, right and left addition is the same
+    __radd__ = __add__
+
+    def __sub__(self, other):
+        """Return ``self - other``."""
+        return self.__add__(-1 * other)
+
+    @property
+    def grad_lipschitz(self):
+        """Lipschitz constant for the gradient of the functional"""
+        return self.__grad_lipschitz
+
+
+class FunctionalLeftScalarMult(Functional, OperatorLeftScalarMult):
+
+    """Scalar multiplication of functional from the left.
+
+    Given a functional ``f`` and a scalar ``scalar``, this represents the
+    functional
+
+        ``(scalar * f)(x) == scalar * f(x)``.
+    """
+
+    def __init__(self, func, scalar):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        func : `Functional`
+            Functional to be scaled.
+        scalar : `float`
+            Number with which to scale the functional.
+        """
+        if not isinstance(func, Functional):
+            raise TypeError('`func` {!r} is not a `Functional` instance.'
+                            ''.format(func))
+
+        scalar = func.range.element(scalar)
+
+        if scalar == 0:
+            Functional.__init__(self, space=func.domain, linear=True,
+                                grad_lipschitz=0)
+
+        else:
+            Functional.__init__(self, space=func.domain,
+                                linear=func.is_linear,
+                                grad_lipschitz=(
+                                    np.abs(scalar) * func.grad_lipschitz))
+
+        OperatorLeftScalarMult.__init__(self, operator=func, scalar=scalar)
+
+    @property
+    def functional(self):
+        """The original functional."""
+        return self.operator
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional."""
+        return self.scalar * self.functional.gradient
+
+    @property
+    def convex_conj(self):
+        """Convex conjugate functional of the scaled functional."""
+        # The helper function only allows positive scaling parameters.
+        # Otherwise it gives an error.
+        return ConvexConjugateFuncScaling(
+            self.functional.convex_conj, self.scalar)
+
+    @property
+    def proximal(self):
+        """Return the proximal factory of the scaled functional.
+
+        See Also
+        --------
+        proximal_zero
+        """
+
+        if self.scalar < 0:
+            raise ValueError('Proximal operator of functional scaled with a '
+                             'negative value {} is not well-defined'
+                             ''.format(self.scalar))
+
+        elif self.scalar == 0:
+            return proximal_zero(self.domain)
+
+        else:
+            return lambda sigma: self.functional.proximal(sigma * self.scalar)
```

Comment:
I would take the time to create a function here so that you can add at least a 1 line doc and so `repr` returns something sensible instead of `Lambda at 634gd64`